extensions/spotify-player/src/utils/index.ts

```diff
@@ -0,0 +1,63 @@
+import { getApplications, showHUD } from "@raycast/api";
+import { getState, getTrack } from "../spotify/applescript";
+import { SpotifyPlayingState } from "../spotify/types";
+
+export interface Preferences {
+  closeWindowOnAction: boolean;
+}
+
+export async function isSpotifyInstalled() {
+  const applications = await getApplications();
+  return applications.some(({ bundleId }) => bundleId === "com.spotify.client");
+}
+
+export function trackTitle(track: SpotifyApi.TrackObjectSimplified): string {
+  return `${track.artists[0].name} – ${track.name}`;
+}
+
+export async function showTrackNotification() {
+  try {
+    const [state, track] = await Promise.all([getState(), getTrack()]);
+
+    const trackName = (() => {
+      if (state.state == SpotifyPlayingState.Playing) {
+        return `${track.name} by ${track.artist}`;
+      } else if (state.state === SpotifyPlayingState.Paused) {
+        return `${track.name} by ${track.artist} (Paused)`;
+      } else {
+        return "Not playing";
+      }
+    })();
+
+    await showHUD("🎧 " + trackName);
+  } catch (err) {
+    console.error(err);
+  }
+}
+
+export function msToHMS(milliseconds: number): string {
```

Comment:
There is a slightly non-obvious change here in that we stop returning time strings without the minute part. For example 24 seconds would be returned as `00:24` with this change which in my opinion is clearer as to what it represents.